src/sentry/ratelimits/cardinality.py

```diff
@@ -289,16 +295,25 @@ def check_within_quotas(
                 for request in requests
             ]
 
-        with self.client.pipeline(transaction=False) as pipeline:
-            pipeline.mget(unit_keys_to_get)
-            # O(self.cluster_shard_factor * len(requests)), assuming there's
-            # only one per-org quota
-            for key in set_keys_to_count:
-                pipeline.scard(key)
+        if self.is_redis_cluster:
```

Comment:
I am wondering if it would be better the implementations based on redis cluster and redis blaster be in separate functions for both check_within_quotas and use_quotas. Then during init you can setup the appropriate function to be used based on the appropriate backend. Then in this code you just call those function callbacks. It makes reasoning about the code a bit simpler.

Reply:
i pulled out this code into a helper class with a somewhat typed interface.